components/kyma-environment-broker/cmd/subscriptioncleanup/cloudprovider/aws.go

```diff
@@ -1,18 +1,126 @@
 package cloudprovider
 
+import (
+	"context"
+	"errors"
+	"log"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/credentials"
+	"github.com/aws/aws-sdk-go-v2/service/ec2"
+	"github.com/aws/aws-sdk-go-v2/service/ec2/types"
+)
+
+const defaultRegion string = "eu-central-1"
+
 type awsResourceCleaner struct {
+	ec2Client   ec2.Client
+	credentials awsCredentialsConfig
 }
 
-type awsConfig struct {
+type awsCredentialsConfig struct {
 	accessKeyID     string
 	secretAccessKey string
 }
 
 func NewAwsResourcesCleaner(secretData map[string][]byte) (ResourceCleaner, error) {
-	return awsResourceCleaner{}, nil
+	awsResourceCleaner := awsResourceCleaner{}
+	awsConfig, err := awsResourceCleaner.toAwsConfig(secretData)
+	if err != nil {
+		return nil, err
+	}
+
+	awsResourceCleaner.credentials = awsConfig
+	awsEC2Client, err := awsResourceCleaner.newAwsEC2Client(awsConfig, defaultRegion)
+	if err != nil {
+		return nil, err
+	}
+
+	awsResourceCleaner.ec2Client = awsEC2Client
+	return awsResourceCleaner, nil
 }
 
 func (ac awsResourceCleaner) Do() error {
-	// TODO: clean up resources if required...
+	all_regions := ac.getAllRegions()
+
+	// Iterate through all regions
+	for _, region := range all_regions.Regions {
+		log.Printf("Switching to region %v", *region.RegionName)
+		newEc2, err := ac.newAwsEC2Client(ac.credentials, *region.RegionName)
+		ac.ec2Client = newEc2
+
+		// Iterate through volumes
+		volumes, err := ac.ec2Client.DescribeVolumes(context.TODO(), &ec2.DescribeVolumesInput{})
+		if err != nil {
+			return err
+		}
+
+		log.Println("Listing Volumes:")
+		for _, volume := range volumes.Volumes {
+			log.Printf("Volume ID: %v", *volume.VolumeId)
+		}
+
+		ac.deleteVolumes()
+	}
+
+	return nil
+}
+
+func (ac awsResourceCleaner) deleteVolumes() error {
+	volumes, err := ac.ec2Client.DescribeVolumes(context.TODO(), &ec2.DescribeVolumesInput{})
+	if err != nil {
+		return err
+	}
+
+	for _, volume := range volumes.Volumes {
+		log.Printf("Deleting volume with id: %v", *volume.VolumeId)
+		if volume.State == types.VolumeStateInUse {
+			log.Printf("Volume is in-use, detaching it first: ")
+			ac.ec2Client.DetachVolume(context.TODO(), &ec2.DetachVolumeInput{
+				VolumeId: volume.VolumeId,
+			})
+		}
```

Comment:
I am thinking if we should error out in this case. A HAP account marked as dirty, which has volumes attached to VM instance? To be on the safe side I would not wipe in this case. 
Erroring out will allow us to notice as alert will fire due to K8S job being failed.